test/test_mail.py

```diff
@@ -154,15 +154,6 @@ def test_summary_mail_only_on_error(self):
         self.assertTrue(
             "Packages that attempted to upgrade:\n 2vcard" in mail_txt)
 
-    def test_mail_on_error_with_warning_in_log(self):
```

Comment:
I have not looked deeper but it appears this change (and the one below with the `re.search()` for the warning revert the fix for https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=703621 - is there an alternative for this? Or was the initial fix bad?

Reply:
Aha, I read up the details about this now and it looks like the initial fix was indeed bad. Thanks for catching this.